src/VecSim/vec_sim_index.h

```diff
@@ -93,19 +84,12 @@ class VecSimIndex : public VecsimBaseObject {
      *
      * @return vector dimension
      */
-    inline size_t getVectorDim() { return this->dim; }
+    virtual size_t getVectorDim() = 0; // { return this->dim; }
```

Comment:
why did you leave those here? who requires VecSimIndex to tell it's dim and metric?